### ISSUE TITLE
 Cannot connect to open ai

### ISSUE THREAD

**Issue by heidongzitugou** [MichelNivard/gptstudio#54]
Hello, thank you very much for providing this useful tool. Everything was normal when I used it according to your tutorial. Today when I used it, it prompted me that the connection timed out, but I can use it normally when I connect to the API in other software. [I don’t know if there is a problem with the tool itself or my network]
Here is my error message：
✔ API key found and matches the expected format.
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached: [api.openai.com] Connection timeout after 10003 ms

**Comment by JamesHWade** [comment]
It seems like the API may be temporarily unavailable. We should add a more descriptive error message in these cases. Please comment here if the error persists.

**Comment by panlongyu** [comment]
me too。。。。
Checking API key using OPENAI_API_KEY environment variable...
✔ API key found and matches the expected format.
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached: [api.openai.com] Connection timeout after 10008 ms

**Comment by heidongzitugou** [comment]
The problem still exists. I can call chatgpt’s api normally in obsidian, but it still times out in Rstudio. At present, I don’t know how to solve this problem from which direction.

**Comment by JamesHWade** [comment]
These are a bit tough to track down. Can you please reinstall that package and run `gptstudio:::simple_api_check()` and share the output?

Are you setting the API in you .Renviron file?

**Comment by heidongzitugou** [comment]
Yes, I have set up my API in the .Renviron file. Then I reinstalled this package and the result of gptstudio:::simple_api_check() is as follows:
gptstudio:::addin_chatgpt()
Checking API key using OPENAI_API_KEY environment variable...
✔ API key found and matches the expected format.
Error in curl::curl_fetch_memory(url, handle = handle) :
Timeout was reached: [api.openai.com] Connection timeout after 10012 ms

**Comment by heidongzitugou** [comment]
Yes, I have set up my API in the .Renviron file. Then I reinstalled this package and the result of gptstudio:::simple_api_check() is as follows:

> gptstudio:::simple_api_check()
Error in curl::curl_fetch_memory(url, handle = handle) : 
  Timeout was reached: [api.openai.com] Connection timeout after 10012 ms

**Comment by JamesHWade** [comment]
I'm stumped. You could try running this code on the console:

``` r
library(httr)
api_key <- Sys.getenv("OPENAI_API_KEY")
response <-
  GET(
    "https://api.openai.com/v1/models",
    add_headers(Authorization = paste0("Bearer ", api_key))
  )
status_code(response)
#> [1] 200
```

<sup>Created on 2023-03-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

If that doesn't worked, I'm not sure how to explain it.
